src/fairseq2/models/family.py

```diff
@@ -452,12 +458,27 @@ def iter_checkpoint(
                 f"`config` must be of type `{self._configs.kls}`, but is of type `{type(config)}` instead."
             )
 
-        return self._do_iter_checkpoint(path, config, gangs, mmap, restrict)
+        if not self._supports_meta:
+            _warn_deprecated(
+                "In fairseq2 v0.12, `ModelFamily.iter_checkpoint` will stop supporting models that cannot be meta-initialized."
+            )
+
+        if self._shard_specs is None:
+            model = self._do_create_model(
+                config, gangs, torch.float16, self._supports_meta
```

Comment:
Where does the default `torch.float16` come from? Does this correctly handle the case of normalization layers that should be kept at fp32?

Reply:
Makes sense, but this looks a bit random when glancing over this code! Hopefully somebody will find this discussion when the need for refactoring arises (or just understands that this is not an important parameter).

Reply:
Ok added some inline comment to make it a bit more clear why we initialize a model here.